frontend/src/__tests__/api/books.test.js

```diff
@@ -0,0 +1,229 @@
+import '@testing-library/jest-dom';
+import { act, renderHook, waitFor } from '@testing-library/react';
+import { MockedProvider } from '@apollo/client/testing';
+import {
+  allBooksQuery,
+  aBookQuery,
+  addBookMutation,
+  removeBookMutation,
+  useGetBooks,
+  useGetBook,
+  useAddBook,
+  useRemoveBook
+} from '../../../api/books';
+
+describe('should test the books api', () => {
```

Comment:
minor nit pick, but typically we name them like this:
`describe('file-name or class')`
then in the `it`'s we say 
`it('should ....')`

any further `describe` blocks would essentially be like this:
`describe('when x condition is met')`
`it('should do this...')`

Reply:
Makes sense. Re-namd to "books api test".